File: 199-BinaryTreeRightSideView/199-BinaryTreeRightSideView.py
# Last updated: 07/04/2026, 11:40:31
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if root==None:
            return []
        qu, res =[], []
        qu.append(root)
        while len(qu)>0:
            lgt = len(qu)
            for i in range(lgt):
                node=qu.pop(0)
                if i==lgt-1:
                    res.append(node.val)
                if node.left:
                    qu.append(node.left)
                if node.right:
                    qu.append(node.right)
        return res



        
        

File: 199-BinaryTreeRightSideView/199-BinaryTreeRightSideView.cpp
// Last updated: 07/04/2026, 11:40:26
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {


public:

    vector<int> rightSideView(TreeNode* root) {
        
        vector<int> ans;
        queue<TreeNode*> q;
        if ( !root) return ans;

        q.push(root);

        while(!(q.empty())){
            int k = q.size();
            int val = -1;
            for( int i = 0;i<k;i++){
                root = q.front();
                if ( i == k-1) ans.push_back(root->val);
                q.pop();
                if ( root->left) q.push(root->left);
                if ( root->right ) q.push(root->right);
            }
            

        }
        return ans;




    }
};